lib/ansible/modules/cloud/amazon/ec2_placement_group.py

```diff
@@ -0,0 +1,194 @@
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ec2_placement_group
+short_description: Create or delete an EC2 Placement Group
+description:
+    - Create an EC2 Placement Group; if the placement group already exists,
+      nothing is done. Or, delete an existing placement group. If the placement
+      group is absent, do nothing.
+version_added: "2.5"
+author: "Brad Macpherson (@iiibrad)"
+options:
+  name:
+    description:
+      - The name for the placement group.
+    required: true
+    default: null
+  state:
+    description:
+      - Create or delete placement group.
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide
+# for details.
+
+# Create a placement group.
+- ec2_placement_group:
+    name: my-cluster
+    state: present
+
+# Delete a placement group.
+- ec2_placement_group:
+    name: my-cluster
+    state: absent
+
+'''
+
+
+RETURN = '''
+placement_group:
+  description: Placement group attributes
+  returned: when state != absent
+  type: complex
+  contains:
+    name:
+      description: PG name
+      type: string
+      sample: my-cluster
+    state:
+      description: PG state
+      type: string
+      sample: "available"
+    strategy:
+      description: PG strategy
+      type: string
+      sample: "cluster"
+
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (boto3_conn,
+                                      ec2_argument_spec,
+                                      get_aws_connection_info,
+                                      HAS_BOTO3)
+try:
+    from botocore.exceptions import (BotoCoreError, ClientError)
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
+
+
+def get_placement_group_details(connection, module):
+    name = module.params.get("name")
+    try:
+        response = connection.describe_placement_groups(
+            Filters=[{
+                "Name": "group-name",
+                "Values": [name]
+            }])
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(
+            e,
+            msg="Couldn't find placement group named [%s]" % name)
+
+    if len(response['PlacementGroups']) != 1:
+        return None
+    else:
+        placement_group = response['PlacementGroups'][0]
+        return {
+            "name": placement_group['GroupName'],
+            "state": placement_group['State'],
+            "strategy": placement_group['Strategy'],
+        }
+
+
+def create_placement_group(connection, module):
+    name = module.params.get("name")
+
+    try:
+        connection.create_placement_group(
+            GroupName=name, Strategy='cluster', DryRun=module.check_mode)
```

Comment:
Don't use DryRun - you're not handling the exception that will be returned.

With check_mode, we usually just don't make the API call at all and return something suitable in the results.

Reply:
Now catching the error correctly

Reply:
That's definitely better, but it might just be simpler to do

```
def create_placement_group(connection, module):
    name = module.params.get("name")
    if module.check_mode:
        module.exit_json(changed=True, placement_group={"name": name, "strategy": 'cluster'})
    try:
        connection.create_placement_group(GroupName=name, Strategy='cluster')
    except (BotoCoreError, ClientError) as e:
        module.fail_json_aws(e, msg="Couldn't create placement group [%s]" % name)
etc...
```
+        connection.create_placement_group(

Reply:
That would be simpler, but at the cost of the permissions checking that DryRun gives. Permissions are a pain in our environment as we're loath to give godmode to our automated processes (or our humans, tbh :-) ).

Reply:
That's fair - permissions are definitely worth managing right!

Reply:
I'm cool with DryRun for this use case - the logic here is pretty simple so catching the exception is plenty sufficient.